cmd/simulacra/simulacra.go

```diff
@@ -178,43 +158,127 @@ func configureFromGCloud(project *string, zone *string) error {
+	if *configPath != "" {
+		return getConfigFromYaml(*configPath)
+	}
+
+	config := Config{Platform: *platform, ConfigFilePath: *opsAgentConfigFile, Project: *project, Zone: *zone, Name: *name,
+		ThirdPartyAppsPath: *thirdPartyAppsPath}
+
+	return config, nil
+
+}
+
+func runCustomScripts(ctx context.Context, vm *gce.VM, logger *logging.DirectoryLogger, scripts []string) error {
+	for _, scriptPath := range scripts {
+		if scriptContent, err := os.ReadFile(scriptPath); err != nil {
+			return err
+		} else {
+			logger.ToMainLog().Printf("Running script from %s", scriptPath)
+			log.Default().Printf("Running script from %s", scriptPath)
+			if _, err := gce.RunScriptRemotely(ctx, logger, vm, string(scriptContent), nil, make(map[string]string)); err != nil {
+				return err
+			}
+			logger.ToMainLog().Printf("Done Running Script from  %s", scriptPath)
+
+		}
+	}
+	return nil
+}
+
+func main() {
+	loggingDir := fmt.Sprintf("/tmp/simulacra-%s", uuid.NewString())
+	logger, err := logging.NewDirectoryLogger(loggingDir)
+	if err != nil {
+		log.Default().Fatalf("Error initializing directory logger %v", err)
+	}
+	log.Default().Printf("Starting Simulacra, Detailed logging can be found in %s directory", loggingDir)
+	ctx := context.Background()
+	config, err := getSimulacraConfig()
+	if err != nil {
+		log.Default().Fatalf("error parsing simulacra config %v", err)
+	}
+
+	if err := configureFromGCloud(&config.Project, &config.Zone); err != nil {
+		log.Default().Fatalf("project and zone must either be non empty or set in GCloud %v", err)
 	}
 
 	options := gce.VMOptions{
-		Platform:    *platform,
-		MachineType: agents.RecommendedMachineType(*platform),
-		Name:        *name,
-		Project:     *project,
-		Zone:        *zone,
+		Platform:    config.Platform,
+		MachineType: agents.RecommendedMachineType(config.Platform),
+		Name:        config.Name,
+		Project:     config.Project,
+		Zone:        config.Zone,
 	}
 	// Create VM Instance.
-	vm, err := gce.CreateInstance(ctx, logger, options)
+	log.Default().Printf("Creating VM Instance, check %s for details", vmInitLogFile)
+	vm, err := gce.CreateInstance(ctx, logger.ToFile(vmInitLogFile), options)
 	if err != nil {
-		logger.Fatalf("Failed to create GCE instance %v", err)
+		log.Default().Fatalf("Failed to create GCE instance %v", err)
 	}
 	// Install Ops Agent on VM.
-	if err := setupOpsAgent(ctx, vm, logger, *configFile); err != nil {
-		logger.Fatalf("Failed to install Ops Agent %v", err)
+	log.Default().Print("Installing Ops Agent, check main_log.txt for details")
+	if err := setupOpsAgent(ctx, vm, logger.ToMainLog(), config.ConfigFilePath); err != nil {
+		log.Default().Fatalf("Failed to install Ops Agent %v", err)
 	}
 
 	// Install Third Party Appliations based on Ops Agent Config.
-	if err := installApps(ctx, vm, logger, *configFile, *thirdPartyAppsPath); err != nil {
-		logger.Fatalf("Failed to install apps %v", err)
+	if len(config.ConfigFilePath) > 0 {
```

Comment:
Why did we need this condition? If we have the default config be used (equivalent to no config specified), we should still run the scripts (installing the apps is a noop since none are detected in the config)

Reply:
Basically what my thinking process was that, if we take out the condition then the function 'installApps' returns with an error "no file with path "" found" (smth like that). Now its not exactly an 'error' since not specifying the config file was intentional. This is why I wanted to only execute the function if configFilePath is actually explicitly specified.

Reply:
oh i see your comment about just short circuiting the function itself, that makes sense :)